internal/trust/remotes.go

```diff
@@ -162,7 +162,11 @@ func (r *Remotes) Replace(dir string, newRemotes ...types.ClusterMember) error {
 			return fmt.Errorf("Failed to parse remote %q to yaml: %w", remote.Name, err)
 		}
 
-		remotePath := filepath.Join(dir, filepath.Base(remote.Name+".yaml"))
+		remotePath, err := filepath.Abs(filepath.Join(dir, remote.Name+".yaml"))
+		if err != nil || !strings.HasPrefix(remotePath, dir) {
```

Comment:
The if block should also be split to prevent printing an nil error `%!w(<nil>)`.